src/plots_masked.py

```diff
@@ -2,86 +2,174 @@
+            nonlocal done
+            done = True
+
+        _cbtn = close_button(close_plot)
+        _kpbtn = keep_button(f_path)
+        _rpbtn = replace_button(f_path, pred)
+        _edit_m_btn = edit_mask_button(f_path)
+        _edit_p_btn = edit_pred_button(f_path, pred)
+        _dltbtn = delete_button(f_path)
+        maximize_plot()
+        plt.show()
+
+
 def plot_predictions(
-    predictions, test_iter: Iterator, dataset: List[str], t_f: bool
+    predictions, test_iter: Iterator
 ) -> None:
     """ Plots the Neural Nets predictions, the mask images and sar images """
-
     done = False
-    for pred, (img, mask), f_path in zip(predictions, test_iter, dataset):
+    for pred, (img, mask) in zip(predictions, test_iter):
         # Plots imgs from img_dict
         if done:
             break
-        plt.subplot(1, 4, 1)
-        plt.title('prediction')
-        plt.imshow(pred.reshape(512, 512), cmap=plt.get_cmap('gist_gray'))
-
-        plt.subplot(1, 4, 2)
-        plt.title('mask')
-        plt.imshow(mask.reshape(512, 512), cmap=plt.get_cmap('gist_gray'))
-
-        plt.subplot(1, 4, 3)
-        plt.title('vh img')
-        img = img.clip(0, 1)
-        plt.imshow(img[0, :, :, 0].reshape(512, 512),
-                   cmap=plt.get_cmap('gist_gray'))
-
-        plt.subplot(1, 4, 4)
-        plt.title('vv img')
-        img = img.clip(0, 1)
-        plt.imshow(img[0, :, :, 1].reshape(512, 512),
-                   cmap=plt.get_cmap('gist_gray'))
+
+        plots(pred, mask, img)
 
         def close_plot(_: Any) -> None:
             nonlocal done
             done = True
 
-        if t_f is True:
-            _cbtn = close_button(close_plot)
-            _kpbtn = keep_button()
-            _rpbtn = replace_button(f_path[2], pred)
-            maximize_plot()
-            plt.show()
-        else:
-            _cbtn = close_button(close_plot)
-            maximize_plot()
-            plt.show()
+        _cbtn = close_button(close_plot)
+        maximize_plot()
+        plt.show()
+
+
+def plots(pred, mask, img) -> None:
+    plt.subplot(1, 4, 1)
+    plt.title('prediction')
+    plt.imshow(pred.reshape(512, 512), cmap=plt.get_cmap('gist_gray'))
+
+    plt.subplot(1, 4, 2)
+    plt.title('mask')
+    plt.imshow(mask.reshape(512, 512), cmap=plt.get_cmap('gist_gray'))
+
+    plt.subplot(1, 4, 3)
+    plt.title('vh img')
+    img = img.clip(0, 1)
+    plt.imshow(img[0, :, :, 0].reshape(512, 512),
+               cmap=plt.get_cmap('gist_gray'))
+
+    plt.subplot(1, 4, 4)
+    plt.title('vv img')
+    img = img.clip(0, 1)
+    plt.imshow(img[0, :, :, 1].reshape(512, 512),
+               cmap=plt.get_cmap('gist_gray'))
+
+
+def save_img(f_paths: List[str], pred) -> None:
+    with gdal_open(f_paths[2]) as f:
+        mask_projection = f.GetProjection()
+        mask_geo_transform = f.GetGeoTransform()
+
+    write_mask_to_file(
+        pred.reshape(512, 512), f_paths[2], mask_projection, mask_geo_transform
```

Comment:
maybe change 512 to a variable to keep code DRY

Reply:
I'll change that in the plots function since reshape is used more than once.